Test/Framework/Graphics/Texture2DNonVisualTest.cs

```diff
@@ -465,5 +465,109 @@ public void SetData5ParameterExceptionTest(int arraySize, int startIndex, int el
+            t.GetData(b2);
+
+            Assert.AreEqual(b, b2);
+
+            t.Dispose();
+        }
+
+        [TestCase(0)]
+        [TestCase(1)]
+        public void GetAndSetDataDxtNotMultipleOf4Rounding(int mip)
+        {
+            var t = content.Load<Texture2D>(Paths.Texture ("random_16px_dxt"));
+
+            var before = new byte[t.Width*t.Height/2];
+            t.GetData(before);
+
+            var b1 = new byte[8];
+            var b2 = new byte[8];
+
+            t.GetData(mip, new Rectangle(0,0,4,4), b1, 0, 8);
+
+            t.GetData(mip, new Rectangle(0,0,1,1), b2, 0, 8);
+            t.SetData(mip, new Rectangle(0,0,1,1), b2, 0, 8);
+            Assert.AreEqual(b1, b2);
+
+            t.GetData(mip, new Rectangle(0,0,1,3), b2, 0, 8);
+            t.SetData(mip, new Rectangle(0,0,1,3), b2, 0, 8);
+            Assert.AreEqual(b1, b2);
+
+            t.GetData(mip, new Rectangle(0,0,4,3), b2, 0, 8);
+            t.SetData(mip, new Rectangle(0,0,4,3), b2, 0, 8);
+            Assert.AreEqual(b1, b2);
+
+            t.GetData(mip, new Rectangle(0, 2, 4, 4), b2, 0, 8);
+            t.SetData(mip, new Rectangle(0, 2, 4, 4), b2, 0, 8);
+            Assert.AreEqual(b1, b2);
+
+            t.GetData(mip, new Rectangle(2, 2, 4, 4), b2, 0, 8);
+            t.SetData(mip, new Rectangle(2, 2, 4, 4), b2, 0, 8);
+            Assert.AreEqual(b1, b2);
+
+            t.GetData(mip, new Rectangle(3, 3, 4, 4), b2, 0, 8);
+            t.SetData(mip, new Rectangle(3, 3, 4, 4), b2, 0, 8);
+            Assert.AreEqual(b1, b2);
+
+            t.GetData(mip, new Rectangle(4, 4, 4, 4), b2, 0, 8);
+            t.SetData(mip, new Rectangle(4, 4, 4, 4), b2, 0, 8);
+            Assert.AreNotEqual(b1, b2);
+
+            var after = new byte[t.Width*t.Height/2];
+            t.GetData(after);
+
+            Assert.AreEqual(before, after);
+
+            t.Dispose();
+        }
+
+        [Test]
+        public void GetAndSetDataDxtDontRoundWhenOutsideBounds()
+        {
+            var t = content.Load<Texture2D>(Paths.Texture("random_16px_dxt"));
+
+            var b = new byte[8];
+
+            // don't round if the unrounded rectangle would be outside the texture area
+            Assert.Throws<ArgumentException>(() => t.GetData(0, new Rectangle(15, 15, 3, 3), b, 0, 8));
+            // this does work
+            t.GetData(0, new Rectangle(15, 15, 1, 1), b, 0, 8);
+
+            t.Dispose();
+        }
+
+        [Test]
+        public void GetAndSetDataDxtLowerMips()
+        {
+            var t = content.Load<Texture2D>(Paths.Texture("random_16px_dxt"));
+
+            var b = new byte[8];
+            var b2 = new byte[8];
+
+            t.GetData(0, new Rectangle(0,0,4,4), b, 0, 8);
+            t.GetData(1, new Rectangle(0,0,4,4), b2, 0, 8);
+            t.GetData(2, new Rectangle(0,0,4,4), b2, 0, 8);
+            t.GetData(3, new Rectangle(0,0,2,2), b2, 0, 8);
+            t.GetData(4, new Rectangle(0,0,1,1), b2, 0, 8);
```

Comment:
@KonajuGames Do you know why this behaves the way it does? You can only pass a one pixel rectangle (larger will cause an exception), yet it fills up 8 bytes. Maybe internally it stores one DXT block with a solid color? This is the last mip btw since the texture is 16x16 px.